tests/integration/test_registry.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# pylint: disable=duplicate-code
+"""Integration tests."""
+
+import json
+import logging
+import random
+import string
+from pathlib import Path
+from typing import List
+
+import pytest
+import yaml
+from juju import model
+from kubernetes.utils import create_from_yaml
+
+from . import helpers
+
+# This pytest mark configures the test environment to use the Canonical Kubernetes
+# bundle with ceph, for all the test within this module.
+pytestmark = [
+    pytest.mark.bundle(file="test_registries/test-bundle-docker-registry.yaml", apps_local=["k8s"])
+]
+
+log = logging.getLogger(__name__)
+
+
+def _get_data_file_path(name) -> Path:
+    """Retrieve the full path of the specified test data file."""
+    return Path(__file__).parent / "data" / "test_registries" / name
```

Comment:
Yep, this can just be  a `CONSTANT`

```suggestion
TEST_DATA_PATH = Path(__file__).parent / "data" / "test_registries" / name
```